### ISSUE TITLE
Hello, can you provide complete code of project?

### ISSUE THREAD

**Issue by NGA-MMDDsteven** [zjk1988/DAN-Net#3]
Thanks for you implentment of Dan-Net. But I am still confused about how to construct a complete project to run this network.
Up to now, you have provided the whole network structure. But the training process and loss function are still not available.

If you are available , I want to have more contact with you about how to build FP and FBP layer in torch and how to prepare the dataset.

（感谢你开源了Dan-Net网络代码，受自身能力有限，目前开源的代码我还无法直接用以网络训练，希望您能提供更完整的代码！如果您方便的话，我也想像您请教FBP以及FP的实现方法，以及您是如何准备训练数据集的）

**Comment by zjk1988** [comment]
Thanks for noticing my work, the sinogram-domain, image-domain and reconstruction layer have been presented. You can train the model the way you like and the environment requires has been listed.
About FBP, we reimplement the reconstruction layer of DuDoNet, but we don't use FP in this work.
About preparing the dataset, you can see the paper [https://iopscience.iop.org/article/10.1088/1361-6560/ac1156](https://iopscience.iop.org/article/10.1088/1361-6560/ac1156).

**Comment by NGA-MMDDsteven** [comment]
> Thanks for noticing my work, the sinogram-domain, image-domain and reconstruction layer have been presented. You can train the model the way you like and the environment requires has been listed.
> About FBP, we reimplement the reconstruction layer of DuDoNet, but we don't use FP in this work.
> About preparing the dataset, you can see the paper https://iopscience.iop.org/article/10.1088/1361-6560/ac1156.

thanks for your reply, I will try to build the database first!

**Comment by RobinvanRiel** [comment]
Thank you for implementing the DAN-Net but I still can't get the correct database. Is it possible to get the Matlab code for this?

**Comment by chenjiachengzzz** [comment]
> > Thanks for noticing my work, the sinogram-domain, image-domain and reconstruction layer have been presented. You can train the model the way you like and the environment requires has been listed.
> > About FBP, we reimplement the reconstruction layer of DuDoNet, but we don't use FP in this work.
> > About preparing the dataset, you can see the paper [https://iopscience.iop.org/article/10.1088/1361-6560/ac1156[  ](https://sci-hub.wf/10.1088/1361-6560/ac1156)](https://iopscience.iop.org/article/10.1088/1361-6560/ac1156).
> 
> thanks for your reply, I will try to build the database first!
您好，您数据集构造成功了嘛？
